controllers/component_build_controller_common.go

```diff
@@ -49,6 +49,10 @@ func getGitProvider(component appstudiov1alpha1.Component) (string, error) {
 	allowedGitProviders := map[string]bool{"github": true, "gitlab": true, "bitbucket": true}
 	gitProvider := ""
 
+	if component.Spec.Source.GitSource == nil {
```

Comment:
if component.Spec.Source.GitSource == nil || component.Spec.Source.GitSource.URL == "" {

would be better

Reply:
Empty URL is handled later, it will return an error anyway